trieste/acquisition/sampler.py

```diff
@@ -186,358 +174,55 @@ def binary_search(val: float) -> float:  # Find empirical interquartile range
         b = (q1 - q2) / (l1 - l2)
         a = (q2 * l1 - q1 * l2) / (l1 - l2)
 
-        uniform_samples = tf.random.uniform([self._sample_size], dtype=fmean.dtype)
+        uniform_samples = tf.random.uniform([sample_size], dtype=fmean.dtype)
         gumbel_samples = log(-log(1 - uniform_samples)) * tf.cast(b, fmean.dtype) + tf.cast(
             a, fmean.dtype
         )
         gumbel_samples = tf.expand_dims(gumbel_samples, axis=-1)  # [S, 1]
         return gumbel_samples
 
 
-class IndependentReparametrizationSampler(Sampler):
+class ThompsonSamplerFromTrajectory(ThompsonSampler):
     r"""
-    This sampler employs the *reparameterization trick* to approximate samples from a
-    :class:`ProbabilisticModel`\ 's predictive distribution as
-
-    .. math:: x \mapsto \mu(x) + \epsilon \sigma(x)
-
-    where :math:`\epsilon \sim \mathcal N (0, 1)` is constant for a given sampler, thus ensuring
-    samples form a continuous curve.
+    This sampler provides approximate Thompson samples of the objective function's
+    minimiser :math:`x^*` by minimizing approximate trajectories sampled from the
+    underlying probabilistic model. This sampling method can be used for any
+    probabilistic model with a :meth:`trajectory_sampler` method.
     """
 
-    def __init__(self, sample_size: int, model: ProbabilisticModel):
-        """
-        :param sample_size: The number of samples to take at each point. Must be positive.
-        :param model: The model to sample from.
-        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
-        """
-        super().__init__(sample_size, model)
-
-        # _eps is essentially a lazy constant. It is declared and assigned an empty tensor here, and
-        # populated on the first call to sample
-        self._eps = tf.Variable(
-            tf.ones([sample_size, 0], dtype=tf.float64), shape=[sample_size, None]
-        )  # [S, 0]
-
-    def sample(self, at: TensorType) -> TensorType:
+    def sample(self, model: ProbabilisticModel, sample_size: int, at: TensorType) -> TensorType:
```

Comment:
if we implement the new interface, then we can be more precise here with `TrajectorySamplerModel`